tools/c7n_mailer/tests/test_azure.py

```diff
@@ -90,3 +92,66 @@ def test_sendgrid_handler(self, mock_send):
         result = sendgrid_delivery.sendgrid_handler(self.loaded_message, sendgrid_messages)
         self.assertTrue(result)
         mock_send.assert_called()
+
+    @patch('c7n_mailer.azure_mailer.azure_queue_processor.SmtpDelivery')
+    def test_smtp_delivery(self, mock_smtp):
+        smtp_mailer_config = {
+            'queue_url': 'asq://storageaccount.queue.core.windows.net/queuename',
+            'from_address': 'you@youremail.com',
+            'smtp_port': 25,
+            'smtp_ssl': True,
+            'smtp_server': 'test_server',
+            'smtp_username': 'user',
+            'smtp_password': 'password'
+        }
+
+        with patch('c7n_mailer.azure_mailer.sendgrid_delivery.SendGridDelivery'
+                   '.get_to_addrs_sendgrid_messages_map',
+                   return_value={('mock@test.com',): self.loaded_message}):
+            azure_processor = MailerAzureQueueProcessor(smtp_mailer_config, logger)
+            self.assertTrue(azure_processor.process_azure_queue_message(self.compressed_message))
+            mock_smtp.assert_has_calls(
+                [call().send_message(message=self.loaded_message, to_addrs=['mock@test.com'])])
+
+    @patch('c7n_mailer.slack_delivery.SlackDelivery')
+    def test_slack_delivery(self, mock_slack):
+        slack_mailer_config = {
+            'queue_url': 'asq://storageaccount.queue.core.windows.net/queuename',
+            'slack_token': 'mock_token'
+        }
+
+        slack_compressed_message = MagicMock()
+        slack_compressed_message.content = base64.b64encode(
+            zlib.compress(ASQ_MESSAGE_SLACK.encode('utf8')))
+        slack_loaded_message = json.loads(ASQ_MESSAGE_SLACK)
+
+        mock_slack.return_value\
```

Comment:
nit: isn't it `mock_slack.get_to_addrs_slack_messages_map.return_value`?

Reply:
No, actually this is what Andy and I learned earlier in the morning. Reason being is the mock_slack isn't the actual object instance that is being mocked. So when it goes to instantiate a new SlackDelivery the return value of that needs to mock it's method. This blog post was helpful: https://medium.com/python-pandemonium/python-mocking-you-are-a-tricksy-beast-6c4a1f8d19b2